R/score_transition_risk.R

```diff
@@ -0,0 +1,112 @@
+#'   companies = emissions_companies,
+#'   co2 = products,
+#'   europages_companies = europages_companies,
+#'   ecoinvent_activities = ecoinvent_activities,
+#'   ecoinvent_europages = ecoinvent_europages,
+#'   isic = isic_name
+#' ) |> unnest_product()
+#'
+#' sector_companies <- read_csv(toy_sector_profile_companies())
+#' scenarios <- read_csv(toy_sector_profile_any_scenarios())
+#'
+#' sector_profile_at_product_level <- profile_sector(
+#'   companies = sector_companies,
+#'   scenarios = scenarios,
+#'   europages_companies = europages_companies,
+#'   ecoinvent_activities = ecoinvent_activities,
+#'   ecoinvent_europages = ecoinvent_europages,
+#'   isic = isic_name
+#' ) |> unnest_product()
+#'
+#' result <- score_transition_risk(emissions_profile_at_product_level, sector_profile_at_product_level)
+#'
+#' result |> unnest_product()
+#'
+#' result |> unnest_company()
+score_transition_risk <-
+  function(emissions_profile_at_product_level,
+           sector_profile_at_product_level) {
+    union_emissions_sector_rows <-
+      get_rows_union_for_common_cols(
+        emissions_profile_at_product_level,
+        sector_profile_at_product_level
+      )
+    trs_emissions <-
+      prepare_trs_emissions(emissions_profile_at_product_level)
+    trs_sector <- prepare_trs_sector(sector_profile_at_product_level)
+
+    trs_product <-
+      full_join_emmissions_sector(trs_emissions, trs_sector) |>
+      create_tr_benchmarks_tr_score() |>
+      select(-c("scenario_year", "benchmark")) |>
+      left_join(
+        union_emissions_sector_rows,
+        by = c("companies_id", "ep_product", "activity_uuid_product_uuid")
+      ) |>
+      relocate(relocate_trs_columns(product_level_trs_ranking_reduction_columns())) |>
+      distinct()
+
+    trs_company <- trs_product |>
+      select(
+        trs_company_columns(),
+        product_level_trs_ranking_reduction_columns()
+      ) |>
+      create_benchmarks_averages() |>
+      select(-product_level_trs_ranking_reduction_columns()) |>
+      relocate(relocate_trs_columns(trs_company_avg_columns())) |>
+      distinct()
+
+    nest_levels(trs_product, trs_company)
+  }
+
+create_tr_benchmarks_tr_score <- function(data) {
+  mutate(
+    data,
+    transition_risk_score = ifelse(
+      is.na(.data$profile_ranking) | is.na(.data$reduction_targets),
+      NA,
+      (.data$profile_ranking + .data$reduction_targets) / 2
+    ),
+    benchmark_tr_score = ifelse(
+      is.na(.data$scenario_year) | is.na(.data$benchmark),
+      NA,
+      paste(.data$scenario_year, .data$benchmark, sep = "_")
+    )
+  )
+}
+
+create_benchmarks_averages <- function(data) {
+  mutate(
+    data,
+    transition_risk_score_avg = mean(.data$transition_risk_score, na.rm = TRUE),
+    reduction_targets_avg = mean(.data$reduction_targets, na.rm = TRUE),
+    profile_ranking_avg = mean(.data$profile_ranking, na.rm = TRUE),
+    .by = c("companies_id", "benchmark_tr_score")
```

Comment:
@kalashsinghal are you sure that it is grouped by `benchmark_tr_score`? The `reduction_targets_avg` and the `profile_ranking_avg` can also be calculated even though the `transition_risk_score` is `NA`. So if you group by the `benchmark_tr_score `and it is NA I understand the code as such as you cannot calculate `reduction_targets_avg `and `profile_ranking_avg`? Is this correct?

Reply:
@AnneSchoenauer Great catch! It seems like an error. In essence, I think `profile_ranking_avg` should be averaged using the `benchmark` column and the `reduction_targets_avg` should be calculated using the `scenario_year` benchmark column in sector profile. Am I right? 

I think I made a mistake by averaging the `profile_ranking` column using `benchmark_tr_score` because it also contains benchmark values from sector profile (ex: `ipr_2030`, `weo_2050`) which is wrong as they are not related to `profile_ranking`. Right?

Reply:
Hi @kalashsinghal. Okay I understood! I think it is more that when you calculate the results on a product level that you need to make sure that you use the right benchmarks. To avoid this I think it would be best that you incorporate the profile_ranking_avg more in the emission_profile functions and the 'reduction_targets_avg' in the sector profile functions. 

I think that the profile_ranking_avg should then be available in the 'emission_profile_company' data file and the reduction_targets_avg should then be availabine in the 'sector_profile_company' data set. 

So overall what you could do is to delete the averaging functions for the transition score overall but then create a new ticket to add them in the emission profile and sector profile indicators. Does this make sense?

Reply:
@AnneSchoenauer Should I add the average columns for the upstream indicators are well for both emission and sector profile?

Reply:
Yes please.